rasa/cli/x.py

```diff
@@ -291,16 +304,76 @@ def rasa_x(args: argparse.Namespace):
         run_locally(args)
 
 
-def run_in_production(args: argparse.Namespace):
-    from rasa.cli.utils import print_success
-    from rasa.core.utils import AvailableEndpoints
+def _pull_runtime_config_from_server(
+    config_endpoint: Optional[Text],
+    attempts: int = 30,
+    wait_time_between_pulls: Union[int, float] = 0.5,
+    keys: Tuple[Text, ...] = ("endpoints", "credentials"),
+) -> Tuple[Text, ...]:
+    """Pull runtime config from `config_endpoint`.
 
-    print_success("Starting Rasa X in production mode... 🚀")
+    Returns a path to yaml dumps, each containing the contents of one of `keys`.
+    """
+
+    while attempts > 0:
+        try:
+            response = requests.get(config_endpoint)
+            if response.status_code == 200:
+                rjs = response.json()
+                return tuple(_dump_dict_to_yaml(rjs, k) for k in keys)
+            else:
+                logger.debug(
+                    "Failed to get a proper response from remote "
+                    "server. Status Code: {}. Response: {}"
+                    "".format(response.status_code, response.text)
+                )
+        except requests.exceptions.ConnectionError as e:
+            logger.debug("Failed to connect to server. Retrying. {}".format(e))
+        time.sleep(wait_time_between_pulls)
+        attempts -= 1
+
+    print_error_and_exit(
+        "Could not fetch runtime config from server at '{}'. "
+        "Exiting.".format(config_endpoint)
+    )
+
+
+def _dump_dict_to_yaml(data: Dict, key: Text) -> Optional[Text]:
+    content = data.get(key)
+    if not content:
+        print_error_and_exit(
+            "Failed to find key '{}' in runtime config "
+            "dictionary. Exiting.".format(key)
+        )
+
+    temp_file = tempfile.NamedTemporaryFile(delete=False).name
+    io_utils.write_yaml_file(content, temp_file)
+    return temp_file
+
+
+def _get_available_endpoints(args: argparse.Namespace) -> "AvailableEndpoints":
     args.endpoints = get_validated_path(
         args.endpoints, "endpoints", DEFAULT_ENDPOINTS_PATH, True
     )
-    endpoints = AvailableEndpoints.read_endpoints(args.endpoints)
-    _rasa_service(args, endpoints)
+    return AvailableEndpoints.read_endpoints(args.endpoints)
+
+
+def run_in_production(args: argparse.Namespace):
+    from rasa.cli.utils import print_success
+
+    print_success("Starting Rasa X in production mode... 🚀")
+
+    config_endpoint = args.config_endpoint
+    if config_endpoint:
```

Comment:
is there a difference between this code for `production` and `local` or can we put that in a function?